datasafe-simple-adapter/datasafe-simple-adapter-impl/src/test/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeAdapterTest.java

```diff
@@ -300,6 +305,40 @@ void testTwoUsers(WithStorageProvider.StorageDescriptor descriptor) {
 
     }
 
+    @ParameterizedTest
+    @MethodSource("minioOnly")
+    @SneakyThrows
+    void testExhaustedInputStream(WithStorageProvider.StorageDescriptor descriptor) {
+        myinit(descriptor);
+        mystart();
+        long openFileHandlesBefore = 0;
+        OperatingSystemMXBean os = ManagementFactory.getOperatingSystemMXBean();
+        if(os instanceof UnixOperatingSystemMXBean){
+            openFileHandlesBefore = ((UnixOperatingSystemMXBean) os).getOpenFileDescriptorCount();
+        } else {
+            log.error("test for open filehandles can not be done");
+            return;
+        }
+
+        String content = "content of document qdm;mwm;ewmfmwemf;we;mfw;emf;llllle";
+        String path = "a/b/c.txt";
+        DSDocument document = new DSDocument(new DocumentFQN(path), new DocumentContent(content.getBytes()));
+        simpleDatasafeService.storeDocument(userIDAuth, document);
+        IntStream.range(0, 10).forEach(it -> {
```

Comment:
I suggest to create Amazon client with i.e. 2 connection pool size. And await for test to finish within 10 seconds, if it does not finish in this time frame - for 10 readers (IntStream.range(0, 10)) this would mean pool was exhausted and it code fails the test